code/game/atoms.dm

```diff
@@ -567,9 +573,13 @@ its easier to just keep the beam vertical.
 		O.show_message(message,2,deaf_message,1)
 
 /atom/Entered(var/atom/movable/AM, var/atom/old_loc, var/special_event)
-	if(loc && MOVED_DROP == special_event)
-		AM.forceMove(loc, MOVED_DROP)
-		return CANCEL_MOVE_EVENT
+	if(loc)
+		for(var/i in AM.contents)
+			var/atom/movable/A = i
+			A.entered_with_container(old_loc)
```

Comment:
What's the purpose of this entered_with_container thing?

Reply:
When item is in container it doesnt call /moved or /entered, this is solution for that

Reply:
if you have better name for this proc do tell